### PR TITLE
Created backToTop button

### LINKED ISSUE #42: Add back to top button on mobile version only
Add back to top button on the left bottom side in all the categories (Home Furniture, Tablewares, Kitchenware, Home decor) and about us page, on mobile version.
Not necessary for the desktop version.

### PR DESCRIPTION
**Pull request by KostasTheodoro**
closes #42

### CODE REVIEW

File: src/components/BackToTop.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import { useEffect, useState } from "react";
+
+export const BackToTop = () => {
+  const [backToTop, setBackToTop] = useState(false);
+
+  useEffect(() => {
+    window.addEventListener("scroll", () => {
```
Comment (oorestisime) [inline src/components/BackToTop.tsx:7]: you need to remove the listener on unmount.

File: src/components/BackToTop.tsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
+import { useEffect, useState } from "react";
```
Comment (oorestisime) [inline src/components/BackToTop.tsx:1]: I think you need to add 'use client' on this component as it needs for window to be available

File: src/components/BackToTop.tsx [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+      behavior: "smooth",
+    });
+  };
+  return (
+    <div>
+      {backToTop && (
+        <button
+          onClick={scrollUp}
+          id="topButton"
+          className="fixed z-10 p-3 bg-brown-primary rounded-full shadow-md bottom-10 left-10 text-white animate-bounce"
```
Comment (oorestisime) [inline src/components/BackToTop.tsx:24]: Shouldn't this be only for mobile? i dont see any media queries

File: src/components/BackToTop.tsx (hunk 1 above)
Comment (oorestisime) [inline src/components/BackToTop.tsx:7]: Also if this is only on mobile you probably only need the listener for mobile only. Unless you include the component only in small screens.

**Review by oorestisime — CHANGES_REQUESTED** [review]

**Review by oorestisime — CHANGES_REQUESTED** [review]
Left some ocmments

File: src/components/BackToTop.tsx (hunk 3 above)
Comment (KostasTheodoro) [reply src/components/BackToTop.tsx]: nope

File: src/components/BackToTop.tsx (hunk 1 above)
Comment (KostasTheodoro) [reply src/components/BackToTop.tsx:7]: it's not only for mobile

**Pull request merged**